### PR TITLE
feat: explicitly specify bundled or file system

### PR DESCRIPTION
**Pull request by doitian**
Eliminate ambiguity and simplify code.

BREAKING CHANGE:

config ckb.toml and ckb-miner.toml

```diff
- spec = "specs/dev.toml"
+ spec = { file = "specs/dev.toml" }
```

chain spec

```diff
[genesis.system_cells]
files = [
-   "specs/cells/secp256k1_blake160_sighash_all"
+   { bundled = "specs/cells/secp256k1_blake160_sighash_all" }
]
```

### CODE REVIEW

**Comment by doitian** [comment]
@nervos-bot @nervos-bot ci-status 97668e0c6d20c5f5b1daf3496189c12579e787be

CI: success ✅
Integration: success ✅

**Comment by doitian** [comment]
@nervos-bot ci-status 97668e0c6d20c5f5b1daf3496189c12579e787be

CI: success ✅
Integration: success ✅

File: resource/ckb.toml [hunk 1]
```diff
@@ -7,11 +7,11 @@ data_dir = "data"
 
 [chain]
 # Choose the kind of chains to run, possible values:
-# - specs/dev.toml
-# - specs/testnet.toml
-spec = "specs/dev.toml" # {{
-# testnet => spec = "specs/testnet.toml"
-# integration => spec = "specs/integration.toml"
+# - { file = "specs/dev.toml" }
+# - { bundled = "specs/testnet.toml" }
```
Comment (jjyr) [inline resource/ckb.toml]: `bundled` is hard to understand in config, can we use some intention related name like `builtin`.

File: resource/src/lib.rs [hunk 2]
```diff
@@ -22,201 +23,134 @@ pub const CKB_CONFIG_FILE_NAME: &str = "ckb.toml";
...
+                file.read_to_end(&mut data)?;
+                Ok(Cow::Owned(data))
+            }
+        }
     }
 
-    pub fn export_miner<'a>(&self, context: &TemplateContext<'a>) -> Result<()> {
-        self.export(MINER_CONFIG_FILE_NAME, context)
+    /// Gets resource input stream
+    pub fn read(&self) -> Result<Box<dyn Read>> {
```
Comment (quake) [inline resource/src/lib.rs]: unused function?

File: resource/src/lib.rs [hunk 3]
```diff
@@ -22,201 +23,134 @@ pub const CKB_CONFIG_FILE_NAME: &str = "ckb.toml";
...
     }
 
-    pub fn export_ckb<'a>(&self, context: &TemplateContext<'a>) -> Result<()> {
-        self.export(CKB_CONFIG_FILE_NAME, context)
+    /// Gets resource content
+    pub fn get(&self) -> Result<Cow<'static, [u8]>> {
+        match self {
+            Resource::Bundled { bundled } => BUNDLED.get(bundled),
+            Resource::FileSystem { file } => {
+                let mut file = BufReader::new(fs::File::open(file)?);
```
[the remark this answers is not in the document]
Comment (doitian) [reply resource/src/lib.rs]: Copied from https://github.com/tilpner/includedir/blob/master/lib/src/lib.rs#L71

File: resource/src/lib.rs (hunk 2 above)
Comment (doitian) [reply resource/src/lib.rs]: It's a library, the library user can use when needed.

File: resource/src/lib.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (quake) [reply resource/src/lib.rs]: I mean the implementation of `read_to_end` is already buffered, no need to wrap it again with `BufReader`.

[the remark this answers is not in the document]
Comment (doitian) [reply resource/src/lib.rs]: I'll just use `std::fs::read` here

**Pull request merged**